### ISSUE TITLE
Wrong vein data being shown and saved on Ross128b

### ISSUE THREAD

**Issue by CatsLeftEar** [GTNewHorizons/GT-New-Horizons-Modpack#18182]
### Your GTNH Discord Username

Keyla3300

### Your Pack Version

stable

### Your Server

Epsilon

### Java Version

Java 8

### Type of Server

None

### Your Expectation

Expected to see Ross128b veins
![image](https://github.com/user-attachments/assets/65901a39-a2b9-4c5b-b583-53b739c7dcb5)

### The Reality

![image](https://github.com/user-attachments/assets/c72e4af7-d552-4519-bf25-27f15731268f)
getting cosmic neutronium, tengam, awa draconium veins and all others which are clearly not on Ross. Underground obviousely vein are normal, which belong on Ross.

### Your Proposal

bugs bad, fixes good

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [X] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by RAFAEL-SOSA-UH** [comment]
try it in a solo world in 2.7 version alpha 4, and couldn't replicate the issue.

**Comment by tunnels-end** [comment]
I've been experiencing similar (perhaps not identical, I don't have enough detail to go on,) but haven't been able to consistently reproduce it. I messed around a lot last night; anecdotally the veins appear normal at first, only appear wonky after completely rebooting mc, and rebooting sometimes fixes them.

I expect prospecting loads veintypesLUT (i.e. the numerical vein encoding) on startup and the specific world information is loaded on world load, so you'll get the wrong veins displaying if the two use different encoding. This would happen if, say, you like me updated the pack, forgot to copy the visualprospecting folder over, and did so later while minecraft was running. This does not explain everything I was experiencing though, at one point both an old and a new prospecting data were displaying wrong.  Maybe veintypesLUT is generated in a nondeterministic way or something?

Basically, OP, were you by any chance messing with the visualprospecting folder while minecraft was running?

**Comment by CatsLeftEar** [comment]
> I've been experiencing similar (perhaps not identical, I don't have enough detail to go on,) but haven't been able to consistently reproduce it. I messed around a lot last night; anecdotally the veins appear normal at first, only appear wonky after completely rebooting mc, and rebooting sometimes fixes them.
> 
> I expect prospecting loads veintypesLUT (i.e. the numerical vein encoding) on startup and the specific world information is loaded on world load, so you'll get the wrong veins displaying if the two use different encoding. This would happen if, say, you like me updated the pack, forgot to copy the visualprospecting folder over, and did so later while minecraft was running. This does not explain everything I was experiencing though, at one point both an old and a new prospecting data were displaying wrong. Maybe veinIds are generated in a nondeterministic way or something? Checking through the repo rn.
> 
> Basically, OP, were you by any chance messing with the visualprospecting folder while minecraft was running?

Never went into folders

**Comment by jua2ja2** [comment]
I faced a similar issue of a mislabeled vein. I'm playing on a server. I went around detecting veins with the ore finding wand, swapping in each ore chunks raw ore until I found the vein for the chunk, and I detected 4 iron veins at once, all near each other (it said "4 veins detected"). I went to one of them, and found an oilsands vein, not an iron vein. I couldn't relabel the vein. I'm not sure how to reproduce this issue.
